src/modules/paymentProcessor/PP_Queue_v1.sol

```diff
@@ -0,0 +1,941 @@
+        delete _unclaimableAmountsForRecipient[client][token][_msgSender()];
+
+        // Make sure to let paymentClient know that amount doesnt have
+        // to be stored anymore.
+        IERC20PaymentClientBase_v1(client).amountPaid(token, amount);
+
+        // Call has to succeed otherwise no state change.
+        IERC20(token).safeTransferFrom(client, paymentReceiver, amount);
+
+        emit TokensReleased(paymentReceiver, address(token), amount);
+    }
+
+    /// @dev    Validates if a queue ID is valid.
+    /// @param  queueId_ The queue ID to validate.
+    /// @return isValid_ True if the queue ID is valid.
+    function _validQueueId(uint queueId_)
+        internal
+        view
+        returns (bool isValid_)
+    {
+        // Queue ID must be less than or equal to total orders
+        // and greater than 0 (we start from 1).
+        return queueId_ > 0 && queueId_ <= _nextOrderId;
+    }
+
+    /// @dev    Gets payment queue details from flags and data.
+    /// @param  flags_ The payment order flags.
+    /// @param  data_ Additional payment order data.
+    /// @return queueId_ The queue ID from the data.
+    /// @return startTime_ Start time if specified.
+    /// @return cliffPeriod_ Cliff period if specified.
+    /// @return endTime_ End time if specified.
+    function _getPaymentQueueDetails(bytes32 flags_, bytes32[] memory data_)
+        internal
+        pure
+        returns (
+            uint queueId_,
+            uint startTime_,
+            uint cliffPeriod_,
+            uint endTime_
+        )
+    {
+        uint dataIndex = 0;
+
+        // Check if orderID flag is set (bit 0)
+        if (uint(flags_) & (1 << FLAG_ORDER_ID) != 0) {
+            if (dataIndex < data_.length) {
+                queueId_ = uint(data_[dataIndex++]);
+            }
+        }
+
+        // Check if start time flag is set (bit 1)
+        if (uint(flags_) & (1 << FLAG_START_TIME) != 0) {
+            if (dataIndex < data_.length) {
+                startTime_ = uint(data_[dataIndex++]);
+            }
+        }
+
+        // Check if cliff period flag is set (bit 2)
+        if (uint(flags_) & (1 << FLAG_CLIFF_PERIOD) != 0) {
+            if (dataIndex < data_.length) {
+                cliffPeriod_ = uint(data_[dataIndex++]);
+            }
+        }
+
+        // Check if end time flag is set (bit 3)
+        if (uint(flags_) & (1 << FLAG_END_TIME) != 0) {
+            if (dataIndex < data_.length) {
+                endTime_ = uint(data_[dataIndex]);
+            }
+        }
+    }
+
+    /// @dev    Validates if a payment order can be processed based on
+    /// its timing constraints.
+    /// @param  orderId_ ID of the order being validated.
+    /// @param  startTime_ Start time of the order.
+    /// @param  cliffPeriod_ Cliff period of the order.
+    /// @param  endTime_ End time of the order.
+    /// @return valid_ True if the order can be processed.
+    function _validPaymentTiming(
```

Comment:
We don't have a need for vesting details here ser, so the validation is also not needed

Reply:
removed